### PR TITLE
Feat/order list mocks

### PR DESCRIPTION
**Pull request by delasous**
Built off of #567 

Adds static mocks for our customer & supplier order list views. Check them out at `/orders/c` and `orders/s`.

## Release Notes

- **New Features**
  - Introduced new components: `PageCenterDialog`, `OrderedTable`, and `UnorderedTable` for enhanced order management.
  - Added `CustomerOrderMetaForm` for managing customer order metadata.
  - Implemented `CheckControl` and `TextControl` for improved form handling.

- **Improvements**
  - Enhanced form validation and handling with `sveltekit-superforms` and `zod`.
  - Updated various components to streamline data handling and improve user experience.

- **Bug Fixes**
  - Minor formatting adjustments across multiple components for consistency and readability.
  
- **Chores**
  - Updated dependencies in `package.json` for improved functionality and performance.

### CODE REVIEW

**Comment by silviot** [comment]
I think it makes sense to hear what @coderabbitai review full has to say about this.

File: apps/web-client/src/lib/forms/BookForm.svelte [hunk 1]
```diff
@@ -32,14 +24,13 @@
...
+	export let onFetch: (isbn: string, form: SuperForm<BookFormSchema>["form"]) => void = () => {};
 
-	const _form = superValidateSync(data, bookSchema);
-	const form = superForm(_form, options);
+	const form = superForm(data, options);
 
 	const { form: formStore, constraints, enhance } = form;
 
-	const priceProxy = numberProxy(formStore, "price", { emptyIfZero: false, empty: "undefined" });
+	const priceProxy = numberProxy(formStore, "price", { empty: "undefined" });
```
[the remark this answers is not in the document]
Comment (delasous) [reply apps/web-client/src/lib/forms/BookForm.svelte:33]: This is incorrect. The empty option expects a string value of either 'null' | 'undefined' | 'zero'

**Review by silviot — APPROVED** [review]
I believe the per-row `Reconcile` buttons in the *ordered* list on https://test.libroc.co/feat/order-list-mocks/orders/s/ should go:

* needing at least two clicks to create the reconciliation object acts as a sort of confirmation
* it's more obvious that you can select multiple orders if there's no action button next to each one

I'm not 100% sure it's the right choice, but I thought to bring this up.

In any case, the views look very good. Approved!

**Comment by delasous** [comment]
> I believe the per-row `Reconcile` buttons in the _ordered_ list on https://test.libroc.co/feat/order-list-mocks/orders/s/ should go:
> 
>     * needing at least two clicks to create the reconciliation object acts as a sort of confirmation
> 
>     * it's more obvious that you can select multiple orders if there's no action button next to each one
> 
> 
> I'm not 100% sure it's the right choice, but I thought to bring this up.
> 
> In any case, the views look very good. Approved!

I agree with this. But I say we hold off on making the tweak until we start to wire things in. I have added mocks for `/orders/s/[id]` - to place an order and `orders/s/reconcile`. There are additional rough points in here. For instance, in the "place order view" I have added some buttons to "select 1/2", "select 1/4". I don't like this but I left it so that we can have a conversation around it. I see these as the same fixes that we can make as we work with these mocks

**Pull request merged**